docs/developers_guide/deploying_spack.rst

```diff
@@ -0,0 +1,482 @@
+test on several compilers automatically.
+
+Testing MALI with Albany
+------------------------
+
+Please use the MALI-Dev submodule in compass for testing, rather than MALI-Dev
+develop branch.  The submodule is the version we know works with compass and
+serves as kind of a baseline for other testing.
+
+.. code-block:: bash
+
+    # source whichever load script is appropriate
+    source load_dev_compass_1.2.0-alpha.5_chrysalis_gnu_openmpi_albany.sh
+    git submodule update --init --recursive
+    cd MALI-Dev/components/mpas-albany-landice
+    # you need to tell it to build with Albany
+    make ALBANY=true gfortran
+    compass suite -s -c landice -t full_integration -p . \
+        -w /lcrc/group/e3sm/ac.xylar/compass/test_20230202/landice_full_chrys_gnu_openmpi
+    cd /lcrc/group/e3sm/ac.xylar/compass/test_20230202/landice_full_chrys_gnu_openmpi
+    sbatch job_script.full_integration.bash
+
+Testing MPAS-Ocean with PETSc
+-----------------------------
+
+The tests for PETSc use nonhydrostatic capabilities not yet integrated into
+E3SM.  So you can’t use the E3SM-Project submodule.  You need to use Sara
+Calandrini’s `nonhydro <https://github.com/scalandr/E3SM/tree/ocean/nonhydro>`_
+branch.
+
+.. code-block:: bash
+
+    # source whichever load script is appropriate
+    source load_dev_compass_1.2.0-alpha.5_chrysalis_intel_openmpi_netlib_lapack_petsc.sh
+    git submodule update --init
+    cd E3SM-Project
+    git remote add scalandr/E3SM git@github.com:scalandr/E3SM.git
+    git worktree add ../nonhydro_chrys_intel_openmpi -b nonhydro_chrys_intel_openmpi \
+        --checkout scalandr/E3SM/ocean/nonhydro
+    cd ../nonhydro_chrys_intel_openmpi
+    git submodule update --init --recursive
+    cd components/mpas-ocean
+    # this will build with scorpio, Netlib LAPACK and PETSc
+    make ifort
+    compass list | grep nonhydro
+    # update these numbers for the 2 nonhydro test cases
+    compass setup -n 245 246 -p . \
+        -w /lcrc/group/e3sm/ac.xylar/compass/test_20230202/nonhydro_chrys_intel_openmpi
+    cd /lcrc/group/e3sm/ac.xylar/compass/test_20230202/nonhydro_chrys_intel_openmpi
+    sbatch job_script.custom.bash
+
+As with non-PETSc MPAS-Ocean and MALI, you can have different worktrees with
+Sara’s nonhydro branch for building with different compilers or use
+`utils/matrix <https://github.com/MPAS-Dev/compass/tree/main/utils/matrix>`_ to
+build (and run).
+
+Deploying shared spack environments
+===================================
+
+.. note::
+
+  Be careful about deploying shared spack environments, as changes you make
+  can affect other compass users.
+
+Once compass has been tested with the spack builds in a temporary location, it
+is time to deploy the shared spack environments for all developers to use.
+A ``mache`` developer will make a ``mache`` release (if needed) before this
+step begins.  So there is no need to build mache from a remote branch anymore.
```

Comment:
Can you add a note explaining how compass knows where to deploy spack to?  Does mache specify a shared deploy location on each machine?

Reply:
A added an explanation that this comes from the compass machine config files (not from mache).